spec/models/king_spec.rb

```diff
@@ -12,4 +12,51 @@
       expect(king.valid_move?(2, 6)).to eq(false)
     end
   end
+
+  describe '#can_castle?' do
+    before :each do
+      @game = FactoryGirl.create(:game)
```

Comment:
I think maybe the issue is that you are not stubbing out populate_board! similar to other test files like piece_spec.rb(line 6). So when you make @game it's populating the board with all the pieces instead of just the pieces you want so castling won't work. So maybe try that and see if it helps.